source/tomopy/prep/stripe.py

```diff
@@ -687,30 +692,31 @@ def _detect_stripe(listdata, snr):
     return listmask
 
 
-def _rs_large(sinogram, snr, size, matindex):
+def _rs_large(sinogram, snr, size, matindex, drop_ratio=0.1, norm=True):
     """
-    Remove large stripes by: locating stripes, normalizing to remove
-    full stripes, using the sorting technique to remove partial stripes.    
+    Remove large stripes.
     """
-    badpixelratio = 0.05
+    drop_ratio = np.clip(drop_ratio, 0.0, 0.8)
     (nrow, ncol) = sinogram.shape
-    ndrop = np.int16(badpixelratio * nrow)
-    sinosorted = np.sort(sinogram, axis=0)
-    sinosmoothed = median_filter(sinosorted, (1, size))
-    list1 = np.mean(sinosorted[ndrop:nrow - ndrop], axis=0)
-    list2 = np.mean(sinosmoothed[ndrop:nrow - ndrop], axis=0)
-    listfact = list1 / list2
+    ndrop = int(0.5 * drop_ratio * nrow)
+    sinosort = np.sort(sinogram, axis=0)
+    sinosmooth = median_filter(sinosort, (1, size))
+    list1 = np.mean(sinosort[ndrop:nrow - ndrop], axis=0)
+    list2 = np.mean(sinosmooth[ndrop:nrow - ndrop], axis=0)
+    listfact = np.divide(list1, list2,
+                         out=np.ones_like(list1), where=list2 != 0)
```

Comment:
Why is it correct to replace division by zero with 1 in this algorithm? Is it because we expect the two averages to be approximately equal?

Reply:
Correct. This is a normalization step, so the ratios are around 1. Explanation is shown here: https://sarepy.readthedocs.io/toc/section3_1/section3_1_3.html